app/celery/service_callback_tasks.py

```diff
@@ -23,10 +24,61 @@
 
 @notify_celery.task(bind=True, name="send-delivery-status", max_retries=5, default_retry_delay=300)
 @statsd(namespace="tasks")
-def send_delivery_status_to_service(self, notification_id):
+def send_delivery_status_to_service(self, notification_id,
+                                    encrypted_status_update=None
+                                    ):
+    if not encrypted_status_update:
+        process_update_with_notification_id(self, notification_id=notification_id)
+    else:
+        try:
+            status_update = encryption.decrypt(encrypted_status_update)
+
+            data = {
+                "id": str(notification_id),
+                "reference": status_update['notification_client_reference'],
+                "to": status_update['notification_to'],
+                "status": status_update['notification_status'],
+                "created_at": status_update['notification_created_at'],
+                "completed_at": status_update['notification_updated_at'],
+                "sent_at": status_update['notification_sent_at'],
+                "notification_type": status_update['notification_type']
+            }
+
+            response = request(
+                method="POST",
+                url=status_update['service_callback_api_url'],
+                data=json.dumps(data),
+                headers={
+                    'Content-Type': 'application/json',
+                    'Authorization': 'Bearer {}'.format(status_update['service_callback_api_bearer_token'])
+                },
+                timeout=60
+            )
+            current_app.logger.info('send_delivery_status_to_service sending {} to {}, response {}'.format(
+                notification_id,
+                status_update['service_callback_api_url'],
+                response.status_code
+            ))
+            response.raise_for_status()
+        except RequestException as e:
+            current_app.logger.warning(
+                "send_delivery_status_to_service request failed for service_id: {} and url: {}. exc: {}".format(
+                    notification_id,
+                    status_update['service_callback_api_url'],
+                    e
+                )
+            )
+            if not isinstance(e, HTTPError) or e.response.status_code >= 500:
+                try:
+                    self.retry(queue=QueueNames.RETRY)
+                except self.MaxRetriesExceededError:
+                    current_app.logger.exception(
+                        'Retry: send_delivery_status_to_service has retried the max num of times')
```

Comment:
you've missed out the unhandled exception logic that i added yesterday, it would be good to keep that in.